trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/CegarLoopFactory.java

```diff
@@ -221,16 +221,25 @@ private NwaCegarLoop<L> createFiniteAutomataCegarLoop(final IUltimateServiceProv
 					errorLocs, proofProducer, services, mTransitionClazz, stateFactoryForRefinement);
 		}
 
-		return switch (mPrefs.getFloydHoareAutomataReuse()) {
-		case EAGER -> new EagerReuseCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
-				proofProducer, services, Collections.emptyList(), rawFloydHoareAutomataFromFile, mTransitionClazz,
-				stateFactoryForRefinement);
-		case LAZY_IN_ORDER -> new LazyReuseCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs,
-				errorLocs, proofProducer, services, Collections.emptyList(), rawFloydHoareAutomataFromFile,
-				mTransitionClazz, stateFactoryForRefinement);
-		case NONE -> new NwaCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
-				proofProducer, services, mTransitionClazz, stateFactoryForRefinement);
-		};
+		switch (mPrefs.getFloydHoareAutomataReuse()) {
+		case EAGER:
+			return new EagerReuseCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
+					proofProducer, services, Collections.emptyList(), rawFloydHoareAutomataFromFile, mTransitionClazz,
+					stateFactoryForRefinement);
+		case LAZY_IN_ORDER:
+			return new LazyReuseCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
+					proofProducer, services, Collections.emptyList(), rawFloydHoareAutomataFromFile, mTransitionClazz,
+					stateFactoryForRefinement);
+		case NONE:
+			if (mPrefs.isParallelCegarLoop()) {
+				return new ParallelNwaCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
+						proofProducer, services, mTransitionClazz, stateFactoryForRefinement);
+			}
+			return new NwaCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
+					proofProducer, services, mTransitionClazz, stateFactoryForRefinement);
+		default:
+			throw new AssertionError("Unknown Setting: " + mPrefs.getFloydHoareAutomataReuse());
```

Comment:
Please do not add a `default` case here.

The previous version of the switch was exhaustive, and the exhaustiveness was checked by the Java compiler. This ensures that if in the future, new enum values are added, we get a compiler error and see immediately which code needs to be adapted. The `default` case undermines this.